src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,656 @@
+/* Repair tile runs the repair protocol for a Firedancer node. */
+
+#define _GNU_SOURCE 
+
+#include "tiles.h"
+
+#include "generated/repair_seccomp.h"
+#include "../../../../flamenco/repair/fd_repair.h"
+#include "../../../../flamenco/fd_flamenco.h"
+#include "../../../../util/fd_util.h"
+#include "../../../../disco/tvu/util.h"
+#include "../../../../disco/fd_disco.h"
+#include "../../../../disco/keyguard/fd_keyload.h"
+#include "../../../../disco/shred/fd_stake_ci.h"
+
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <linux/unistd.h>
+#include <sys/random.h>
+#include <netdb.h>
+#include <netinet/in.h>
+
+#include "../../../../util/net/util.h"
+
+#define NET_IN_IDX      0
+#define CONTACT_IN_IDX  1
+#define STAKE_IN_IDX    2
+#define STORE_IN_IDX    3
+
+#define STORE_OUT_IDX   0
+
+#define MAX_REPAIR_PEERS 40200UL
+
+#define SMAX    (1UL << 30UL)
+#define SDEPTH  (1UL << 11UL)
+
+
+static int
+fd_pubkey_eq( fd_pubkey_t const * key1, fd_pubkey_t const * key2 ) {
+  return memcmp( key1->key, key2->key, sizeof(fd_pubkey_t) ) == 0;
+}
+
+static ulong
+fd_pubkey_hash( fd_pubkey_t const * key, ulong seed ) {
+  return fd_hash( seed, key->key, sizeof(fd_pubkey_t) ); 
+}
+
+static void
+fd_pubkey_copy( fd_pubkey_t * keyd, fd_pubkey_t const * keys ) {
+  memcpy( keyd->key, keys->key, sizeof(fd_pubkey_t) );
+}
+
+/* Contact info table */
+#define MAP_NAME     fd_contact_info_table
+#define MAP_KEY_T    fd_pubkey_t
+#define MAP_KEY_EQ   fd_pubkey_eq
+#define MAP_KEY_HASH fd_pubkey_hash
+#define MAP_KEY_COPY fd_pubkey_copy
+#define MAP_T        fd_contact_info_elem_t
+#include "../../../../util/tmpl/fd_map_giant.c"
```

Comment:
Not sure you need to use fd_map_giant here? Do you need persistent key location? I think fd_map_dynamic is probably better?

Reply:
This isn't even used it turns out...
removing